packages/ag-charts-community/src/util/listeners.ts

```diff
@@ -45,6 +52,21 @@ export class Listeners<EventType extends string, EventHandler extends Handler> {
         }
     }
 
+    /**
+     * Ensure the event is always dispatched. If there are no listeners ready, save the event as pending and dispatch
+     * to listeners as soon as they are added.
+     */
+    public dispatchAlwaysResolve(eventType: EventType, ...params: Parameters<EventHandler>): void {
+        if (this.getListenersByType(eventType).length === 0) {
+            this.pendingDispatches.push({ eventType, params });
+            return;
+        }
+
+        this.pendingDispatches = this.pendingDispatches.filter((pending) => pending.eventType !== eventType);
+
+        this.dispatch(eventType, ...params);
+    }
```

Comment:
And worth highlighting this change, which handles the situation when `zoom.buttons` is set before the toolbar has added it's listener to `proxy-group-options`.